File: app/src/main/java/com/example/sercanyusuf/boomsetdemo/BoomsetApp.java
package com.example.sercanyusuf.boomsetdemo;

import android.app.Application;

import com.example.sercanyusuf.boomsetdemo.di.components.AppComponent;
import com.example.sercanyusuf.boomsetdemo.di.components.DaggerAppComponent;
import com.example.sercanyusuf.boomsetdemo.di.modules.AppModule;


public class BoomsetApp extends Application {

    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }

    public AppComponent getAppComponent(){
        return appComponent;
    }
}

File: app/src/main/java/com/example/sercanyusuf/boomsetdemo/BoomsetApp.kt
package com.example.sercanyusuf.boomsetdemo

import android.app.Application

import com.example.sercanyusuf.boomsetdemo.di.components.AppComponent
import com.example.sercanyusuf.boomsetdemo.di.components.DaggerAppComponent
import com.example.sercanyusuf.boomsetdemo.di.modules.AppModule


class BoomsetApp : Application() {

    val component: AppComponent by lazy {
        DaggerAppComponent
                .builder()
                .appModule(AppModule(this))
                .build()
    }

    override fun onCreate() {
        super.onCreate()
        component.inject(this)
    }
}

